sphinx/screens/chats/chat-common/chat-common/src/main/java/chat/sphinx/chat_common/ui/ChatViewModel.kt

```diff
@@ -807,6 +813,38 @@ abstract class ChatViewModel<ARGS: NavArgs>(
         }
     }
 
+    open fun goToLightningNodePubKeyDetailScreen(url: String?) {
+        viewModelScope.launch(mainImmediate) {
+            if (url?.isValidLightningNodePubKey == true) {
+                url.toLightningNodePubKey()?.let { lightningNodePubKey ->
+                    lightningNodePubKey.getPubKey()?.let { nnPubKey ->
+                        contactRepository.getContactByPubKey(nnPubKey).collect { contact ->
+                            if (contact == null) {
+                                chatNavigator.toAddContactDetail(lightningNodePubKey)
+                            } else {
+                                chatRepository.getConversationByContactId(contact.id).collect { chat ->
+                                    chatNavigator.toChat(chat, contact.id)
+                                }
+                            }
+                        }
+                    }
+                }
+            } else if (url?.isValidTribeJoinLink == true) {
+                url.toTribeJoinLink()?.let { tribeJoinLink ->
```

Comment:
calling `if (.....isValid)` + immediately calling the extension function is not necessary as validity is already checked in the wrapper class' `init` block; the extension function will return null in that case.

```kotlin
if (url != null) {
    url.toLightningNodePubKey()?.let { pubKey ->
        // ... do something
    } ?: url.toTribeJoinLink()?.let { tribeLink ->
        // ... do something
    }
}
```